espnet2/tts/feats_extract/dio.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2020 Nagoya University (Tomoki Hayashi)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""F0 extractor using DIO + Stonemask algorithm."""
+
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+import humanfriendly
+import numpy as np
+import pyworld
+import torch
+import torch.nn.functional as F
+
+from scipy.interpolate import interp1d
+from typeguard import check_argument_types
+
+from espnet.nets.pytorch_backend.nets_utils import pad_list
+from espnet2.tts.feats_extract.abs_feats_extract import AbsFeatsExtract
+
+
+class Dio(AbsFeatsExtract):
+    """F0 estimation with dio + stonemask algortihm."""
```

Comment:
- Maybe you can add more reference information about this (sorry if you put it somewhere)?
- It's better to state that this part cannot be a part of the computational graph and add TODO somewhere about the possibility of rewriting the f0 estimation like that.
- Do you think we can use it for ASR? If so, we'll test it.

Reply:
> Maybe you can add more reference information about this (sorry if you put it somewhere)?

Sure!

> It's better to state that this part cannot be a part of the computational graph and add TODO somewhere about the possibility of rewriting the f0 estimation like that.

Agree. I will add a description about it.

> Do you think we can use it for ASR? If so, we'll test it.

It can be if the task is relatively clean.
But in speech synthesis, we usually adjust the range of f0 for each speaker.
In the case of ASR, it is difficult. So I'm not sure what is happened.

Reply:
> > Do you think we can use it for ASR? If so, we'll test it.
> 
> It can be if the task is relatively clean.
> But in speech synthesis, we usually adjust the range of f0 for each speaker.
> In the case of ASR, it is difficult. So I'm not sure what will be happened.

I think the unadjusted range would be working as a sort of considering the speaker/environment characteristics, and it would be good.
I want to use a pythonic pitch implementation, so I'll try it.